### PR TITLE
File input and dropzone atom

### PR DESCRIPTION
**Pull request by zachstence**
https://flowbite.com/docs/forms/file-input/

https://user-images.githubusercontent.com/31896377/200934733-90733234-7960-43f3-83a6-c7640c64d8e6.mov

### CODE REVIEW

**Comment by 0c370t** [comment]
Task linked: [CU-2q4p7r0 All Form Atoms](https://app.clickup.com/t/2q4p7r0)

File: clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        class:disabled
+        use:filedrop={{multiple, disabled}}
+        on:filedrop={onFiledrop}
+        on:filedragenter={() => (isDropping = true)}
+        on:filedragleave={() => (isDropping = false)}
+    >
+        <!-- Hidden input only for screenreaders -->
+        <input id={labelId} type="file" aria-describedby={error ? errorId : undefined} {disabled} />
+
+        {#if render === "input"}
```
Comment (0c370t) [inline clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte:112]: Move the condition out of the variable and into the markdown

File: clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        value?: FileInputValue<Multi>;
+        multiple: Multi;
+        showDropzone?: FileInputShowDropzone;
+
+        size?: FormControlSize;
+        placeholder?: string;
+        disabled?: boolean;
+        state?: FormControlState;
+        error?: string;
+    }
```
Comment (0c370t) [inline clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte:30]: Remove Generics; just return an array of 1 for single file inputs

File: clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+            }
+        } else {
+            inputStatusText = placeholder || "No file chosen";
+            dropzoneStatusText = multiple ? "Drop files here" : "Drop file here";
+        }
+    }
+
+    const onFiledrop = (e: CustomEvent<FileDropSelectEvent>): void => {
+        const {accepted, rejected} = e.detail.files;
+        console.log({accepted});
```
Comment (0c370t) [inline clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte:91]: Console log

File: clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+    .dropzone {
+        @apply w-full h-fit
+            flex items-center justify-center
+            border-2 border-dashed rounded-lg
+            transition-colors;
+
+        .default-dropzone {
+            @apply flex flex-col justify-around;
+        }
+    }
```
Comment (0c370t) [inline clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte:173]: Pull out static classes

**Review by 0c370t — CHANGES_REQUESTED** [review]

File: clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte (hunk 1 above)
Comment (zachstence) [reply clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte:112]: Logic is complex enough that it warrants keeping it in `<script>`

File: clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte (hunk 2 above)
Comment (zachstence) [reply clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte]: [e4cad06](https://github.com/SprocketBot/sprocket/pull/387/commits/e4cad0672213f36589c5ea6fe05caf7799ab3fd3)

File: clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte (hunk 3 above)
Comment (zachstence) [reply clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte]: [28b1eda](https://github.com/SprocketBot/sprocket/pull/387/commits/28b1edaf82bfab8d503e4675bc7d9abd8963a457)

File: clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte (hunk 4 above)
Comment (zachstence) [reply clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte]: [2e048ca](https://github.com/SprocketBot/sprocket/pull/387/commits/2e048ca57d2682d04a5636537719bea1914bddb8)

**Pull request merged**